src/WebAPI/BasicAPIFeatures.cs

```diff
@@ -514,6 +515,22 @@ public static async Task<JObject> SetParamEdits(Session session, JObject rawData
         return new JObject() { ["success"] = true };
     }
 
+    public static async Task<JObject> RegisterTabs(Session session, JObject rawData)
```

Comment:
the heck are clients registering tabs for? That's... backwards. Server should define them.

Reply:
Because of this:
https://github.com/mcmonkeyprojects/SwarmUI/blob/d611e91b7df2eb1c91e7213367bf7c89430195db/src/Pages/Text2Image.cshtml#L27-L31

unless i am mistaken extensions can add arbitrary extra `<li>` elements (or any html actually) to the tab list via this, and there's no server side registry of what's added - that variable is just An String. Enforcing that extensions provide some kind of ExtensionTab record rather than An String would be an API break and I ain't about to go parsing arbitrary extension-provided HTML in server side code

if i *am* mistaken/didn't look hard enough and this html is generated somewhere from an object rather than being arbitrary shit provided by an estension that would be great

Reply:
Yes that's entirely generated, `WebServer#GatherExtensionPageAdditions` generates that, and knows exactly what titles/ids are in use.

Reply:
well that'll teach me to go modifying an unfamiliar codebase at 1:30am. 

ok lemme just. go right ahead and make this less insane